observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemDefinition.java

```diff
@@ -0,0 +1,170 @@
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode("batch"))
+            .setAllowedValues(ALLOWED_SPAN_PROCESSORS)
+            .setValidator(new StringAllowedValuesValidator(ALLOWED_SPAN_PROCESSORS))
+            .build();
+
+    public static final SimpleAttributeDefinition BATCH_DELAY = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.BATCH_DELAY, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(5000))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_QUEUE_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_QUEUE_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(2048))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_EXPORT_BATCH_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_EXPORT_BATCH_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(512))
+            .build();
+
+    public static final SimpleAttributeDefinition EXPORT_TIMEOUT = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.EXPORT_TIMEOUT, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(30000))
+            .build();
+
+    public static final SimpleAttributeDefinition SAMPLER = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SAMPLER_TYPE, ModelType.STRING, true)
+            .setAllowExpression(true)
+            .setXmlName(OpenTelemetryConfigurationConstants.TYPE)
+            .setAttributeGroup(GROUP_SAMPLER)
+            .setRestartAllServices()
+            .build();
+
+    public static final SimpleAttributeDefinition SAMPLER_ARG = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SAMPLER_ARG, ModelType.DOUBLE, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SAMPLER)
+            .setRestartAllServices()
+            .build();
+
+    public static final AttributeDefinition[] ATTRIBUTES = {
+            SERVICE_NAME, EXPORTER, ENDPOINT, SPAN_PROCESSOR_TYPE, BATCH_DELAY, MAX_QUEUE_SIZE, MAX_EXPORT_BATCH_SIZE,
+            EXPORT_TIMEOUT, SAMPLER, SAMPLER_ARG
+    };
+
+    public static final OpenTelemetrySubsystemDefinition INSTANCE = new OpenTelemetrySubsystemDefinition();
```

Comment:
ResourceDefinition instances are only needed during resource registration.  Retaining references to them via static references prevents their garbage collection.

Reply:
Done.